lib/bindings/src/breez_sdk_liquid.udl

```diff
@@ -313,35 +324,59 @@ dictionary GetInfoResponse {
 };
 
 dictionary PrepareSendRequest {
-    string invoice;
+    string payment_destination;
+    u64? amount_sat;
+};
+
+[Enum]
+interface SendDestination {
+    BIP21(LiquidAddressData address_data);
+    Bolt11(LNInvoice invoice);
 };
 
 dictionary PrepareSendResponse {
-    string invoice;
+    SendDestination payment_destination;
     u64 fees_sat;
 };
 
+dictionary SendPaymentRequest {
+    PrepareSendResponse prepare_response;
+};
+
 dictionary SendPaymentResponse {
     Payment payment;
 };
 
+enum ReceiveMethod {
+    "BIP21",
+    "Bolt11",
+};
+
 dictionary PrepareReceivePaymentRequest {
-    u64 payer_amount_sat;
+    u64? payer_amount_sat;
+    bool use_lightning;
 };
 
 dictionary PrepareReceivePaymentResponse {
-    u64 payer_amount_sat;
+    u64? payer_amount_sat;
+    bool use_lightning;
     u64 fees_sat;
 };
 
 dictionary ReceivePaymentRequest {
-    PrepareReceivePaymentResponse prepare_res;
+    PrepareReceivePaymentResponse prepare_response;
     string? description = null;
 };
 
+[Enum]
+interface ReceiveDestination {
```

Comment:
I am wondering if we should have this struct or keep it simple and return a string in the response.
No strong opinion. No strong opnion but since all of the possible results are strings then I am not sure what is the benefit of returning an enum.

Reply:
Two NITs:

1) The three options can be reduced to two: `Bolt11` and `BIP21`, since `BIP21` can also cover a pure Liquid address.

2) In case of using Lightning (swaps), technically there is also a `BIP21` string that gets created in the MRH and can be used by onchain wallets as well. So the two options are more like `LightningAndOnchain` and `OnchainOnly` (exact naming TBD). This means if LN is chosen, we can return both the invoice and the BIP21 (decoded from the MRH).

For example like so
```
interface ReceiveDestination {
    OnchainOnly(string bip21):
    LightningAndOnchain(string invoice, string bip21);
}
```

Then it would help callers if we return this struct, even if it has only strings, e.g. for UIs that can alternate between "pay this invoice" and "pay onchain".